### PR TITLE
Charles/fasta input

### PR DESCRIPTION
**Pull request by cdebourcy**
Allow input files to be fasta. We'll need this to process all the new background samples we've just received

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,293 +10,293 @@
 #
 # It's strongly recommended that you check this file into your version control system.
```
Comment (cdebourcy) [inline db/schema.rb]: Huge diff because the indentations on every line differ. How strange for an autogenerated file

Comment (cdebourcy) [reply db/schema.rb]: Actually this made me realize we don't even need to add type column to input_files. Will change it to just a method and keep schema out of the diff

File: aws_batch/pipeline.py [hunk 2]
```diff
@@ -555,16 +554,16 @@ def run_sample(sample_s3_input_path, sample_s3_output_path,
                          "stats_file": os.path.join(result_dir, STATS_OUT)}
 
     # Download fastqs
-    command = "aws s3 ls %s/ |grep fastq.gz" % (sample_s3_input_path)
+    command = "aws s3 ls %s/ | grep %s" % (sample_s3_input_path, file_type)
```
Comment (boris-dimitrov) [inline aws_batch/pipeline.py]: this is ok if filetype is one of (fasta, fastq.gz) but won't discern between fastq and fastq.gz

Comment (boris-dimitrov) [reply aws_batch/pipeline.py]: also... ```fastanimation.fastq.gz```

may be just `grep '\.%s$'`

Comment (cdebourcy) [reply aws_batch/pipeline.py]: Thanks!

File: app/models/input_file.rb [hunk 3]
```diff
@@ -36,4 +36,8 @@ def s3_source_check
   def file_path
     File.join(sample.sample_path, 'fastqs', name)
   end
+
+  def type
```
Comment (yunfangjuan) [inline app/models/input_file.rb]: type seems to be a sensitive reserve word. Use file_type instead?

File: app/models/input_file.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@ class InputFile < ApplicationRecord
   SOURCE_TYPE_LOCAL = 'local'.freeze
   SOURCE_TYPE_S3 = 's3'.freeze
 
-  FILE_REGEX = %r{\A[^\s\/]+\.fastq.gz}
+  FILE_REGEX = %r{\A[^\s\/]+\.(fastq.gz|fasta)\z}
```
Comment (yunfangjuan) [inline app/models/input_file.rb]: and |fastq|fasta.gz?

File: aws_batch/pipeline.py [hunk 5]
```diff
@@ -1152,6 +1177,7 @@ def main():
     global STAR_GENOME
     global BOWTIE2_GENOME
     INPUT_BUCKET = os.environ.get('INPUT_BUCKET', INPUT_BUCKET)
+    FILE_TYPE = os.environ.get('FILE_TYPE', FILE_TYPE)
```
Comment (yunfangjuan) [inline aws_batch/pipeline.py]: I think you need to add 
global FILE_TYPE to be able to change the FILE_TYPE

Comment (cdebourcy) [reply aws_batch/pipeline.py]: thanks

File: app/models/input_file.rb (hunk 4 above)
Comment (cdebourcy) [reply app/models/input_file.rb]: thanks!

File: aws_batch/pipeline.py (hunk 5 above)
Comment (cdebourcy) [reply aws_batch/pipeline.py]: thanks!

File: app/models/input_file.rb (hunk 3 above)
Comment (cdebourcy) [reply app/models/input_file.rb]: ok

File: app/models/sample.rb [hunk 6]
```diff
@@ -48,11 +48,16 @@ def sample_path
   validates_associated :input_files
 
   def input_files_checks
-    # validate that we have exactly 2 input files
-    errors.add(:input_files, "file_size !=2 for sample") unless input_files.size == 2
-    # validate that both input files have the same source_type
+    # validate that we have the correct number of input files
+    if host_genome && host_genome.name == HostGenome::NO_HOST_NAME
+      errors.add(:input_files, "no input files") unless input_files.size >= 1
```
Comment (yunfangjuan) [inline app/models/sample.rb]: have to be between 1 and 2.

**Pull request merged**